### PR TITLE
Implement Husky + lint-staged for pre-commit hooks

### LINKED ISSUE #40: Add pre-commit hooks for better code quality control
### Feature details?

[Git hooks](https://git-scm.com/docs/githooks) using frameworks like [husky](https://typicode.github.io/husky/#/) + [lint-staged](https://github.com/okonet/lint-staged) or [pre-commit](https://pre-commit.com/) will ensure bad coding standards & practices aren't introduced to the source code. They'll cleaned & taken care of right at the source.

Further quality control can be ensured by implementing an automated CI/CD pipeline but for starters, Git hooks is the way to go.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by tarunkishore2303**
## Description

Added husky + lint-staged configs for pre-commit checks

Fixes #40 

## Changes Include

- [x] Refactor ⚒️ (code change that neither fixes a bug nor adds a feature)
- [x] Tooling ⚙️ (Changes to the build process or auxiliary tools and libraries such as documentation generation)
- [x] Performance ⚡ (code change that improves performance)- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by tarunkishore2303** [comment]
I've tried to create a fresh pull request as I have made many mistakes in previous one.

**Comment by afkcodes** [comment]
LGTM, just include the .husky folder. please test it on your local make a dummy commit and hard reset it after checking if it works.

**Comment by tarunkishore2303** [comment]
> LGTM, just include the .husky folder. please test it on your local make a dummy commit and hard reset it after checking if it works.

I have made the additions you requested and it is working perfectly fine.

This is what I get when I commit. 
image

**Comment by afkcodes** [comment]
don't push the commits just make it on local

**Comment by tarunkishore2303** [comment]
> don't push the commits just make it on local

Okay fine ! My bad

**Comment by afkcodes** [comment]
seems like husky is running let others have a look.

File: package.json [hunk 1]
```diff
@@ -7,13 +7,25 @@
     "build": "next build",
     "start": "next start"
   },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "src/**/*.{ts,css,html}": [
```
Comment (Jarmos-san) [inline package.json:16]: Could you rewrite this section to something like:

```diff
"lint-staged": {
-  "src/**/*.{ts,css,html}": [
+    "src/**/*.{tsx,ts}": [
     "next lint,
+    "prettier --write"
   ],
+  "**/*.{json,md,yml}": "prettier --write"
},
```

Everything else is :100: right now, thanks for recreating a fresh PR.

Explanation about the changes I requested:

1. We're using TypeScript which is why we need to lint `.tsx` & `.ts` files.
2. Prettier will standardise un-uniform coding practices at at "commit-time".

cc: @varshney-himanshu could you confirm the `prettier` commands we implemented here are correct?

**Review by Jarmos-san — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (tarunkishore2303) [reply package.json:16]: Sure will do !

**Comment by tarunkishore2303** [comment]
Changes done ! Please do have a look

**Review by tarunkishore2303 — COMMENTED** [review]
Changes done ! Please have a look to confirm

**Review by Jarmos-san — APPROVED** [review]

**Comment by Jarmos-san** [comment]
@all-contributors please add @tarunkishore2303 for code

**Pull request merged**